src/main/java/org/fogbowcloud/manager/core/datastore/orderstorage/AttachmentOrderStorage.java

```diff
@@ -0,0 +1,163 @@
+package org.fogbowcloud.manager.core.datastore.orderstorage;
+
+import org.apache.log4j.Logger;
+import org.fogbowcloud.manager.core.datastore.commands.AttachmentSQLCommands;
+import org.fogbowcloud.manager.core.exceptions.UnexpectedException;
+import org.fogbowcloud.manager.core.models.linkedlists.SynchronizedDoublyLinkedList;
+import org.fogbowcloud.manager.core.models.orders.AttachmentOrder;
+import org.fogbowcloud.manager.core.models.orders.Order;
+import org.fogbowcloud.manager.core.models.orders.OrderState;
+import org.fogbowcloud.manager.core.models.tokens.FederationUser;
+
+import java.sql.*;
+import java.util.Date;
+import java.util.Map;
+
+public class AttachmentOrderStorage extends OrderStorage {
+
+    private static final Logger LOGGER = Logger.getLogger(AttachmentOrderStorage.class);
+
+    public AttachmentOrderStorage() throws SQLException {
+        Statement statement = null;
+        Connection connection = null;
+
+        try {
+            connection = getConnection();
+
+            statement = connection.createStatement();
+
+            statement.execute(AttachmentSQLCommands.CREATE_ATTACHMENT_ORDER_TABLE_SQL);
+
+            statement.close();
+        } catch (SQLException e) {
+            LOGGER.error("Error creating attachment order table", e);
+            throw e;
+        } finally {
+            closeConnection(statement, connection);
+        }
+    }
+
+    public void addOrder(Order order) {
+        AttachmentOrder attachmentOrder = (AttachmentOrder) order;
+
+        Connection connection = null;
+        PreparedStatement orderStatement = null;
+
+        try {
+            connection = getConnection();
+            connection.setAutoCommit(false);
+
+            orderStatement = connection.prepareStatement(AttachmentSQLCommands.INSERT_ATTACHMENT_ORDER_SQL);
+
+            addOverallOrderAttributes(orderStatement, attachmentOrder);
+
+            orderStatement.setString(8, attachmentOrder.getSource());
+            orderStatement.setString(9, attachmentOrder.getTarget());
+            orderStatement.setString(10, attachmentOrder.getDevice());
+            orderStatement.setTimestamp(11, new Timestamp(new Date().getTime()));
+
+            orderStatement.executeUpdate();
+
+            connection.commit();
+        } catch (SQLException e) {
+            LOGGER.error("Couldn't add the attachment order.", e);
+            try {
+                if (connection != null) {
+                    connection.rollback();
+                }
+            } catch (SQLException e1) {
+                LOGGER.error("Couldn't rollback transaction.", e1);
+            }
```

Comment:
you missed the thrown

Reply:
See: https://github.com/fogbow/fogbow-manager-core/pull/75#discussion_r201310553